src/qt3utils/applications/controllers/wavemeter_controller.py

```diff
@@ -0,0 +1,87 @@
+import abc
+import ctypes
+import logging
+
+
+class WavemeterController(abc.ABC):
+    """
+    Base class for other types of wavemeter controllers to inherit from
+    """
+    def __init__(self, logger_level):
+        self.logger = logging.getLogger(__name__)
+        self.logger.setLevel(logger_level)
+
+    @abc.abstractmethod
+    def open(self):
+        """
+        Override this method to open and initialize wavemeter
+        """
+        pass
+
+    @abc.abstractmethod
+    def read_wavemeter(self):
+        """
+        Override this method to read the value from the wavemeter
+        """
+        pass
+
+    @abc.abstractmethod
+    def close_wavemeter(self):
+        """
+        Override this method to close the connection to the wavemeter
+        """
+        pass
+
+    @abc.abstractmethod
+    def configure(self):
+        """
+        Override this method to configure the wavemeter from a dict set via yaml file
+        """
+        pass
+
+
+class WavemeterDllController(WavemeterController):
+    """
+    Class for interfacing with wavemeter hardware
+    """
+    def __init__(self, logger_level, dll_path=""):
+        super(WavemeterDllController, self).__init__(logger_level)
+        if not dll_path == "":
+            self.open(dll_path)
+        self.dll_path = dll_path
```

Comment:
why not store the dll_path in self so that you don't have to change the "open" method signature from its abstract signature?